### PR TITLE
feat: Add `conda` package

### PR DESCRIPTION
**Pull request by ajschmidt8**
This PR adds a `conda` package for `dfg`.

It also updates the GHAs workflows to accommodate building the `conda` and `wheel` packages in separate jobs (to ensure clean environments).

Since [semantic-release](https://github.com/semantic-release/semantic-release) is intended to deploy release artifacts from a single machine, this required updating the release workflow to do the following:

- Run semantic-release with the `--dry-run` flag first. The `--dry-run` flag enables the workflow to determine whether the most recent changes warrant a release.
- If the most recent changes warrant a release, run the build/test workflow. This builds the `conda` and `wheel` packages on separate machines to ensure build isolation. The artifacts are then uploaded with [actions/upload-artifact](https://github.com/actions/upload-artifact/).
- Once the build/test workflow is complete, the final release job downloads the artifacts with [actions/download-artifact](https://github.com/actions/download-artifact) and deploys them with semantic-release.

### CODE REVIEW

**Comment by ajschmidt8** [comment]
Things to do/check after this is merged:

- [ ] Ensure packages are published to PyPI and Anaconda.org correctly
- [ ] Ensure links in PR release comments are correct (e.g. [these](https://github.com/rapidsai/dependency-file-generator/pull/69#issuecomment-1986094492))
- [ ] Ensure automated commits work correctly after release (e.g. [these](https://github.com/rapidsai/dependency-file-generator/commit/a257aed51b617918f307b71969d9a9c4731585a1))
- [ ] Ensure CI images still get triggered correctly after successful release (e.g. [this workflow](https://github.com/rapidsai/ci-imgs/actions/workflows/push.yaml))
- [ ] Update required checks for this repository to reflect new job names

**Review by KyleFromNVIDIA — CHANGES_REQUESTED** [review]

File: ci/output-release-version.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+set -euo pipefail
+
+RELEASE_VERSION="${1}"
+
+echo "release_version=${RELEASE_VERSION}" | tee --append "${GITHUB_OUTPUT:-/dev/stdout}"
```
[the remark this answers is not in the document]
Comment (ajschmidt8) [reply ci/output-release-version.sh]: fixed in [f9358f6](https://github.com/rapidsai/dependency-file-generator/pull/73/commits/f9358f6928aaa5e70412be0ead3d92ee6996ddeb)

File: .github/actions/semantic-release/action.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        PYPI_TOKEN: ${{ inputs.PYPI_TOKEN }}
+        ANACONDA_TOKEN: ${{ inputs.ANACONDA_TOKEN }}
+        DRY_RUN: ${{ inputs.DRY_RUN }}
+      run: |
+        DRY_RUN_FLAG=""
+        if [[ "${DRY_RUN}" == "true" ]]; then
+          DRY_RUN_FLAG="--dry-run"
+        fi
+        npm install
+        npx semantic-release "${DRY_RUN_FLAG}"
```
[the remark this answers is not in the document]
Comment (ajschmidt8) [reply .github/actions/semantic-release/action.yaml]: fixed in [6fc13f9](https://github.com/rapidsai/dependency-file-generator/pull/73/commits/6fc13f924cc73d0421a549dabed1195caf47666f).

[the remark this answers is not in the document]
Comment (ajschmidt8) [reply .github/actions/semantic-release/action.yaml]: and [33a7273](https://github.com/rapidsai/dependency-file-generator/pull/73/commits/33a7273b6cb2bdb534a79b578789f4c6737b7fd6)

**Comment by ajschmidt8** [comment]
currently working through some issues with the release workflow

**Comment by ajschmidt8** [comment]
I pushed a few more commits to fix the release workflow:

- 22b570f8359ae0833fc660dbb6cfe53deb11ac76
- aead21452aa86e5b62526f6091352c0e1780d0ea
- 54f8ed9631ed8df739cfdba68cab1ffbb0c253e7
- d3dda8adcbb645f19f74f5f0263ea73fbb0da958

The release worked fine after those changes:

- https://github.com/rapidsai/dependency-file-generator/releases/tag/v1.12.1
- https://pypi.org/project/rapids-dependency-file-generator/1.12.1/
- https://anaconda.org/rapidsai/rapids-dependency-file-generator/files?version=1.12.1

Unfortunately a release comment wasn't posted on this PR due to an initial issue in the `1.12.0` release, but I have no reason to believe that PR comments won't work correctly for future releases.

**Pull request merged**